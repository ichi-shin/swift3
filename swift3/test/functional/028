#!/bin/bash

# Upload Part Copy

. ./common

S3USER=tester

echo data > $tmp.data

_s3_put /bucket
_s3_put /bucket/source -T $tmp.data

upload_id=$(_s3_post /bucket/object?uploads | \
    _xpath '/InitiateMultipartUploadResult/UploadId/text()')
_s3_put /bucket/object?uploadId=${upload_id}\&partNumber=1 -D - \
    -H 'x-amz-copy-source: /bucket/source' \
    -H "x-amz-copy-source-if-unmodified-since: Wed, 1 Jan 3000 00:00:00 GMT" \
    | _filter_curl xml

_sw_get /bucket+segments | _filter_curl | sed -e "s/${upload_id}/UPLOAD_ID/g"
