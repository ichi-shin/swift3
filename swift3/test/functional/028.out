QA output created by 028
> s3curl --id tester -- -X PUT http://SWIFT_HOST/bucket... 200
> s3curl --id tester -- -X PUT -T /TMP.data http://SWIFT_HOST/bucket/source... 200
> s3curl --id tester -- -X POST http://SWIFT_HOST/bucket/object?uploads... 200
> s3curl --id tester -- -X PUT -D - -H x-amz-copy-source: /bucket/source -H x-amz-copy-source-if-unmodified-since: Wed, 1 Jan 3000 00:00:00 GMT http://SWIFT_HOST/bucket/object?uploadId=UPLOAD_ID&partNumber=1... 200
HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: text/html; charset=UTF-8
Date: DATE
ETag: "ETAG"
Last-Modified: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

<?xml version='1.0' encoding='UTF-8'?>
<CopyPartResult>
  <LastModified>DATE</LastModified>
  <ETag>ETAG</ETag>
</CopyPartResult>
> curl -X GET http://SWIFT_HOST/v1/TENANT/bucket+segments... 200
object/UPLOAD_ID
object/UPLOAD_ID/1
